src/lexer/regexFactory.ts

```diff
@@ -76,6 +76,33 @@ export const parameter = (paramTypes: string[], pattern: string): RegExp | undef
   return patternToRegex(`(?:${typesRegex})(?:${pattern})`);
 };
 
+const buildQStringPatterns = () => {
+  const specialDelimiterMap = {
+    '<': '>',
+    '[': ']',
+    '(': ')',
+    '{': '}',
+  };
+
+  // base pattern for special delimiters, left must correspond with right
+  const singlePattern = "{left}(?:[^{right}]|{right}(?!'))+{right}";
+
+  // replace {left} and {right} with delimiters, collect as array
+  const patternList = Object.entries(specialDelimiterMap).map(([left, right]) =>
+    singlePattern.replace(/{left}/g, escapeRegExp(left)).replace(/{right}/g, escapeRegExp(right))
+  );
+
+  // standard pattern for common delimiters, ignores keys of specialDelimiterMap
+  const standardDelimiterPattern = String.raw`(?<tag>[^\s${escapeRegExp(
+    Object.keys(specialDelimiterMap).join('')
+  )}])(?:(?!\k<tag>).|\k<tag>(?!'))+\k<tag>`;
```

Comment:
I think it would be simpler to just hard-code the list of `<{({` inside regex:

```suggestion
  const standardDelimiterPattern = String.raw`(?<tag>[^\s<[({])(?:(?!\k<tag>).|\k<tag>(?!'))+\k<tag>`;
```

Alternatively I'd suggest moving the construction of this character list out of the regex string, which is complex enough without multiple function calls in the middle.